### PR TITLE
Use atomic counter for message seqnos

### LINKED ISSUE #52: PubSub sometimes failing
[C# test](https://github.com/richardschneider/net-ipfs-api/blob/master/test/CoreApi/PubSubApiTest.cs#L97) and [JS test](https://github.com/ipfs/interface-ipfs-core/blob/master/src/pubsub.js#L496) are intermittently failing.  Both are  publishing multiple messages and then counting the subscribed messages. 

It appears that some published messages are not being relayed to the subscribed channels. In the C# test both publisher and subscriber are on the same peer.

NOTE: these tests are running on Windows.  This was first detected in https://github.com/ipfs/js-ipfs-api/pull/648

### PR DESCRIPTION
**Pull request by vyzo**
Closes #52 

The seqno is now the concatenation of the timestamp and an atomic counter; it's 128-bit wide.

### CODE REVIEW

**Comment by vyzo** [comment]
so I don't know what's the deal with the jenkins tests: it says "passed but marked as unstable". 
What is this supposed to mean?

File: floodsub.go [hunk 1]
```diff
@@ -414,8 +418,10 @@ func (p *PubSub) GetTopics() []string {
 
 // Publish publishes data under the given topic
 func (p *PubSub) Publish(topic string, data []byte) error {
-	seqno := make([]byte, 8)
-	binary.BigEndian.PutUint64(seqno, uint64(time.Now().UnixNano()))
+	seqno := make([]byte, 16)
+	counter := atomic.AddUint64(&p.counter, 1)
+	binary.BigEndian.PutUint64(seqno[:8], uint64(time.Now().UnixNano()))
```
Comment (whyrusleeping) [inline floodsub.go]: If we initially set `counter` to `time.Now().UnixNano()`, then we don't need to get the current time every time we send a message. 

Not a big deal, but might save us some syscalls

Comment (whyrusleeping) [reply floodsub.go]: It also saves us from having to send a whole extra 8 bytes per message

Comment (vyzo) [reply floodsub.go]: good points, nice small efficiencies. i'll open a new PR.

Comment (vyzo) [reply floodsub.go]: actually, there is no syscall involved in getting the time; but still.

Comment (vyzo) [reply floodsub.go]: opened follow-up PR.

**Pull request merged**